docs/design/coreclr/botr/clr-abi.md

```diff
@@ -752,3 +752,42 @@ The return value is handled as follows:
 4. All other cases require the use of a return buffer, through which the value is returned.
 
 In addition, there is a guarantee that if a return buffer is used a value is stored there only upon ordinary exit from the method. The buffer is not allowed to be used for temporary storage within the method and its contents will be unaltered if an exception occurs while executing the method.
+
+# Control Flow Guard (CFG) support on Windows
+
+Control Flow Guard (CFG) is a security mitigation available in Windows.
+When CFG is enabled, the operating system maintains data structures that can be used to verify whether an address is to be considered a valid indirect call target.
+This mechanism is exposed through two different helper functions, each with different characteristics.
+
+The first mechanism is a validator that takes the target address as an argument and fails fast if the address is not an expected indirect call target; otherwise, it does nothing and returns.
+The second mechanism is a dispatcher that takes the target address in a non-standard register; on successful validation of the address, it jumps directly to the target function.
+Windows makes the dispatcher available only on ARM64 and x64, while the validator is available on all platforms.
+However, the JIT supports CFG only on ARM64 and x64, wth CFG by default being disabled for these platforms.
```

Comment:
```suggestion
However, the JIT supports CFG only on ARM64 and x64, with CFG by default being disabled for these platforms.
```

Do we expect to enable it by default in the future, or for some configurations? This file is not about product plans, but if you say it's disabled by default, maybe it's worthwhile saying where it will be expected to be used?

Reply:
We don't expect to enable it by default, I believe the primary use is by NativeAOT for certain constrained environments where CFG is required.